pulsar-transformations/src/test/java/com/datastax/oss/pulsar/functions/transforms/ComputeStepTest.java

```diff
@@ -225,6 +222,106 @@ void testAvro() throws Exception {
     assertEquals(read.get("age"), new Utf8("43"));
   }
 
+  @Test
+  void testAvroInferredType() throws Exception {
```

Comment:
However we handle expression that evaluates to null, it has to be covered with a unit test (I think it is fair to throw an exception as mentioned in the comment).